src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/StatelessDataController.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Threading.Tasks;
+
+using Altinn.App.Api.Filters;
+
+using Altinn.App.Common.Serialization;
+using Altinn.App.PlatformServices.Extensions;
+using Altinn.App.Services.Interface;
+using Altinn.Platform.Storage.Interface.Models;
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace Altinn.App.Api.Controllers
+{
+    /// <summary>
+    /// The stateless data controller handles creation and calculation of data elements not related to an instance.
+    /// </summary>
+    [AutoValidateAntiforgeryTokenIfAuthCookie]
+    public class StatelessDataController : ControllerBase
+    {
+        private readonly ILogger<DataController> _logger;
+        private readonly IAltinnApp _altinnApp;
+        private readonly IAppResources _appResourcesService;
+        private readonly IPrefill _prefillService;
+
+        private const long REQUEST_SIZE_LIMIT = 2000 * 1024 * 1024;
+
+        /// <summary>
+        /// The stateless data controller is responsible for creating and updating stateles data elements.
+        /// </summary>
+        /// <param name="logger">The logger</param>
+        /// <param name="altinnApp">The app logic for current application</param>
+        /// <param name="appResourcesService">The apps resource service</param>
+        /// <param name="prefillService">A service with prefill related logic.</param>
+        public StatelessDataController(
+            ILogger<DataController> logger,
+            IAltinnApp altinnApp,
+            IAppResources appResourcesService,
+            IPrefill prefillService)
+        {
+            _logger = logger;
+
+            _altinnApp = altinnApp;
+            _appResourcesService = appResourcesService;
+            _prefillService = prefillService;
+        }
+
+        /// <summary>
+        /// Create a new data object of the defined data type
+        /// </summary>
+        /// <param name="dataType">The data type id</param>
+        /// <returns>Return a new instance of the data object including prefill and initial calculations</returns>
+        [Authorize]
+        [HttpPost]
+        [DisableFormValueModelBinding]
+        [RequestSizeLimit(REQUEST_SIZE_LIMIT)]
+        [ProducesResponseType(typeof(DataElement), 200)]
+        [Route("{org}/{app}/v1/data")]
```

Comment:
Consider moving this to class level as this is the base route for the controller, then we avoid duplication on each action.